---
title: "1a_Analysis"
author: "Esther"
date: "2024-04-29"
output: html_document
---

```{r}


# upload the necessary packages: make sure to install the packages first
library(DESeq2) # differential gene expression analysis packet from BiocManager
library(dplyr) 
library(radiant)
library(tidyverse)
library(pheatmap)
```

```{r}
# Read the data from the text file
po_data <- read.delim("mirna_filtered_final.txt", header = TRUE, row.names = NULL )
po_data <- po_data[1:11] %>%
  `row.names<-`(., NULL) %>%
   column_to_rownames(var = "miRNA")

# Rename column names
new_column_names <- c("Sham1", "Sham2", "Sham3", "Sham4", "Sham5", "PO1", "PO2", "PO3", "PO4", "PO5")
colnames(po_data) <- new_column_names

```


```{r}
# making the coldata for DeSeq2

# subset desired columns for analysis: 
po_data <- subset(po_data, select = c("Sham1", "Sham2", "Sham3", "Sham4", "Sham5", "PO1", "PO2", "PO3", "PO4", "PO5"))

# making the matrix - there are 10 total samples. 
colData = matrix(c(1:10), ncol = 1, byrow = TRUE) 

# the condition is the drug, so that is what we will want in the DeSeq object.
colnames(colData) = "Surgery"

# make it a dataframe for easier management. 
colData <- as.data.frame(colData)

# assigning the rownames for coldata - they should all match the column names in countdata. 
rownames(colData) <- c("Sham1", "Sham2", "Sham3", "Sham4", "Sham5", "PO1", "PO2", "PO3", "PO4", "PO5")

# adding in the genotypes - what the sample ID's correspond to.
colData$Surgery = c("Sham1", "Sham2", "Sham3", "Sham4", "Sham5", "PO1", "PO2", "PO3", "PO4", "PO5")

# check your dataframe. 
View(colData)
```

```{r}
# Create a vector with the group names for each sample
group_names <- c(rep("SHAM", 5), rep("PO", 5))

# Create a factor variable with the group names
group <- factor(group_names, levels = c("SHAM", "PO"))

```


```{r}
# Create DESeqDataSet object
dds <- DESeqDataSetFromMatrix(countData = po_data, colData = data.frame(group), design = ~ group)
```

```{r}
# Normalize the data
dds <- DESeq(dds)

# Contrast between PO+PDE5-I and PO
contrast <- c("group", "PO", "SHAM")


# Perform differential expression analysis
results <- results(dds, contrast = contrast)

# Summary of results
summary(results)

# Order results by p-value
results <- results[order(results$padj),]

write.csv(results, "deseq2_results_PO.csv", row.names = TRUE)
```

```{r}

# Filter significant results

significant_results <- subset(results, padj < 0.05) 


# View the significant results
head(significant_results)

write.csv(significant_results, "deseq2_sign.resultsV3.csv", row.names = TRUE)
```
```{r}
Volcano_PF9613_PO.png<- "Volcano_PO-PF9613_PO.png"
  # Volcano plot
par(mfrow=c(1,1))
# Define colors based on different conditions
colors <- ifelse(results$log2FoldChange > 1 & results$padj < 0.05, "black",
                 ifelse(results$log2FoldChange < -1 & results$padj < 0.05, "darkgreen",
                        ifelse(results$log2FoldChange > 1, "purple",
                               ifelse(results$log2FoldChange < -1, "red", "blue"))))
# Plot the volcano plot
plot(results$log2FoldChange, -log10(results$pvalue), pch=20, main="Volcano plot of miRs altered in PO versus sham",
     xlim=c(-3,3), ylim=c(0,30), xlab="log2FoldChange", ylab="-log10(pvalue)", col=colors)
# Create legend labels with counts
legend_labels <- c("Significant upregulated", "Significant downregulated", 
                            "Upregulated", "Downregulated", "Non-significant")
# Add legend with adjusted position
legend("topright", legend=legend_labels,
       col=c("black", "darkgreen", "purple", "red", "blue"), pch=20,
       inset=c(0.05, 0.05), box.col = "transparent") # Adjust the inset values to position the legend
# Save the plot as a PNG file Volcano_PO-PF9613_PO.png
dev.copy(png, Volcano_PF9613_PO.png)
dev.off()
```



```{r}
# Recreating the heatmap:
normalized_dds <- varianceStabilizingTransformation(dds, blind = FALSE)
rlog_out <- rlog(dds, blind = FALSE) # transformed with rlog and varianceStabilizingTransformation - rlog looked a little better. 

# Quality check step (choosing which transformation): 
plotPCA(rlog_out, intgroup = "group") #use this one. 

# Build a results table comparing miRNA expression between AGAT deficient and wildtype conditions: 
res_heatmap <- results(dds, contrast = c("group", "SHAM", "PO"))

# Apply the FDR correction (multiple testing)
res_heatmap$p_val_fdr <- p.adjust(res_heatmap$pvalue, method = "fdr")

# Get rid of NA's in order to plot: 
sigs <- na.omit(res_heatmap)

# Subset significant results, based on FDR: 
sigs <- sigs[sigs$p_val_fdr < 0.05,]

# Make sigs into a dataframe: 
sigs <- as.data.frame(sigs)

# Extract counts for significant miRNAs
mat <- counts(dds, normalized = TRUE)[rownames(sigs),]

# Z scale: make the heatmap more informative and interpretable 
mat.z <- t(apply(mat, 1, scale))

# Add desired column names: 
colnames(mat.z) <- c("Sham1", "Sham2", "Sham3", "Sham4", "Sham5", "PO1", "PO2", "PO3", "PO4", "PO5")

# Make heatmap
library(gplots)
mycol <- colorpanel(1000, low = "yellow", high = "red")

heatmap.2(mat.z, dendrogram = "column", key = TRUE,
          col = mycol, density.info="none", trace = "none",
          margins=c(5,10))

          
```

